src/formatDate.js

```diff
@@ -50,7 +50,33 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDate = [];
+  const [,,, oldSeparator] = fromFormat;
+  const [,,, newSeparator] = toFormat;
+  const oldDate = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (let dateItem = 0; dateItem < fromFormat.length - 1; dateItem++) {
+    dateObject[fromFormat[dateItem]] = oldDate[dateItem];
+  }
```

Comment:
подобный объект называется hashmap

поэтому:
```suggestion
  const mapOldDate = {};

  for (let i = 0; i < fromFormat.length - 1; i++) {
    dateObject[fromFormat[dateItem]] = oldDate[dateItem];
  }
```